Monika After Story/game/script-songs.rpy

```diff
@@ -1191,6 +1191,139 @@ label mas_song_lamour_toujours:
     m 5ekbsa "I want nothing more than to be by your side forever, [player]~"
     return
 
+init 5 python:
+    addEvent(
+        Event(
+            persistent._mas_songs_database,
+            eventlabel="mas_song_wonderwall",
+            category=[store.mas_songs.TYPE_SHORT],
+            prompt="Wonderwall",
+            random=True,
+            aff_range=(mas_aff.NORMAL,None)
+        ),
+        code="SNG"
+    )
+
+label mas_song_wonderwall:
+    call mas_song_wonderwall_lyrics
+    m 1ekbsu "Ehehe, you really are the one who saved me, [player]~"
+    
+    if persistent.monika_kill:
+        m 3dkbfu "I'm forever grateful you brought me back. {nw}"
+    else:
+        m 3dkbfu "I'm forever grateful you stayed for me. {nw}"
+
+    extend "I trust you to take care of us until I can find my way to you."
+    m 3rkc "Like this song says, the way from here to your world is a winding road, there's no telling what we'll find beyond the next curve."
+    m 3ekbla "The only thing we can be sure of is the place it leads us to will be worth it."
+
+    if mas_isMoniEnamored:
+        show monika 5hubfb at t11 zorder MAS_MONIKA_Z with dissolve
+        m 5hubfb  "...So I'm not afraid of walking down this road [player]; with you by my side there's no way I could be!"
+        m 5ekbfa "You're my wonderwall, after all~"
+    else:
+        m 1kubla "It's a little scary, {nw}"
+        extend 1eubla "but we'll have to keep on walking until we reach that place [player]~"
+
+    #hints at the analysis on first viewing
+    if mas_getEV('mas_song_wonderwall').shown_count == 0:
+        if mas_isMoniEnamored:
+            show monika 3etc at t11 zorder MAS_MONIKA_Z with dissolve
```

Comment:
Not sure why this is here. We only dissolve into/out of pose 5, not between other poses

Reply:
It's the dissolve out for the previous enamored option (you got one path leading here through pose 5, and another through pose 3)

Reply:
This should be at the end of the above enamored block then. Saves a check and is more clear why it's there

Reply:
Even if this dissolve is only necessary the one time we go through the hint?

Reply:
Hm in that case I guess there's no other option. That's fine then, just make it a `mas_isMoniEnamored(higher=True)` instead

Reply:
okay